### DISCUSSION TITLE
Configurable Content-Layer Config File

### DISCUSSION THREAD

**Discussion by dustinlacewell** [withastro/roadmap#1006] [Proposal]
# Summary

Currently, the content layer uses `content/config.ts` for configuration. Make this configurable.

# Background & Motivation

With the new Content Layer able to pull content from anywhere, it is no longer a valid assumption that the `content/` folder will be utilized for anything but `config.ts`. It can feel a little silly having `content/` around just to hold `config.ts`.

# Goals

Add some sort of configuration to `astro.config.mjs` to allow customizing which file is used by the content layer for configuration. 

# Example

```ts
export default defineConfig({
  content: {
    config: 'src/content.ts`
  }
});
```

**Comment by louiss0** [comment]
> # Summary
> Currently, the content layer uses `content/config.ts` for configuration. Make this configurable.
> 
> # Background & Motivation
> With the new Content Layer able to pull content from anywhere, it is no longer a valid assumption that the `content/` folder will be utilized for anything but `config.ts`. It can feel a little silly having `content/` around just to hold `config.ts`.
> 
> # Goals
> Add some sort of configuration to `astro.config.mjs` to allow customizing which file is used by the content layer for configuration.
> 
> # Example
> ```ts
> export default defineConfig({
>   content: {
>     config: 'src/content.ts`
>   }
> });
> ```

Please put the config file in the root please?

**Comment by ascorbic** [comment]
tbh I'd rather just move it to the root by default, instead of making it configurable.

**Comment by vadimkantorov** [comment]
IMO it's useful to be able to specify paths to the configs (could be even passed in command-line arguments to `astro dev` or via env var). Or allow custom loader of content config to be defined in the main config.

In some set-ups it's useful to be able to place all configs of a project in one directory (or even into a single file). But also, yes, letting the content config to be discoverable in root would be useful too...

**Comment by michaeljolley** [comment]
This would be a great addition to those of us pulling in `src/content` directories via git submodules. I have several Astro sites sharing the same content repo as a submodule. I'd love to keep my config with my content to align my schemas with the actual content. Then pull in that submodule and specify in `astro.config.mjs` where it can find the `content.config.ts`.

**Reply by Fryuni** [reply]
Content Collection config has been moved to `src` instead of `src/content`. It is now read from `src/content.config.ts`. If you want to define the collections inside of the submodule you can simply re-export it like this:

```ts
// In your `src/content.config.ts`
export {collections} from './content/config.ts';
```

**Reply by vadimkantorov** [reply]
In my usecase, I'm trying to put all site configs into a separate directory, so letting the content config path be configurable would be of great help to me

**Reply by Fryuni** [reply]
The same workaround also applies to what you described:

```ts
export {collections} from '../your-separate-directory/config-for-content.ts';
```

**Reply by michaeljolley** [reply]
> ```ts
> xport {collections} from './
> ```

Dude. That's a banger idea. Can't believe I didn't think of that. Thanks!
